### DISCUSSION TITLE
Recaptcha

### DISCUSSION THREAD

**Discussion by countrymusicfy** [AzuraCast/AzuraCast#4226] [Ideas]
Any plans to integrate Recaptcha into login page?

**Comment by ItsMitchh** [comment]
A not bad idea, I'll consider looking into this and seeing how viable it is to implement into AzuraCast.

**Reply by Vaalyn** [reply]
I would personally prefer [hCaptcha](https://www.hcaptcha.com/) instead of reCAPTCHA in order to preserve privacy of users as much as possible. Also a captcha service will need to be optional since AzuraCast should be able to function completely independent from the internet if the user wants it.

I would recommend creating a feature request for this over at https://features.azuracast.com/

**Reply by BusterNeece** [reply]
Yep, what @Vaalyn said. We'd prefer an open and privacy-oriented solution if implemented, and it would be fully optional since we do intend to be fully standalone out of the box. This is why we currently implement rate limiting and intentionally "slow" password hashing as means to help mitigate the issue without leaning on captchas.
